modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/resource/test/HeadlessBuilderResourceTest.java

```diff
@@ -339,6 +339,56 @@ public void testGetWithRequestFilter() throws Exception {
 			JSONCompareMode.LENIENT);
 	}
 
+	@Test
+	public void testPutByExternalReferenceCode() throws Exception {
+		_addAPIApplication(
+			_API_APPLICATION_ERC_1, _API_ENDPOINT_ERC_1, _BASE_URL_1,
+			_objectDefinition1.getExternalReferenceCode(),
+			_API_APPLICATION_PATH_1);
+
+		_publishAPIApplication(_API_APPLICATION_ERC_1);
+
+		_assertSuccessfulHttpCode(
+			HTTPTestUtil.invokeToHttpCode(
+				JSONUtil.put(
+					"applicationStatus", "unpublished"
+				).put(
+					"baseURL", "test"
+				).put(
+					"title", "title"
+				).toString(),
+				"headless-builder/applications/by-external-reference-code/" +
+					_API_APPLICATION_ERC_1,
+				Http.Method.PUT));
+	}
+
+	@Test
+	public void testPutById() throws Exception {
```

Comment:
here we could make it published and then check if when calling the endpoint with the new baseURL we receive a proper response of the endpoint actually being exposed and working

Reply:
we can also add an assert that the old URL does not work anymore

Reply:
Updated, thanks!